packages/loot-core/src/shared/schedules.js

```diff
@@ -42,39 +42,47 @@ export function getHasTransactionsQuery(schedules) {
     .select(['schedule', 'date']);
 }
 
-function makeNumberSuffix(num) {
-  // Slight abuse of date-fns to turn a number like "1" into the full
-  // form "1st" but formatting a date with that number
-  return monthUtils.format(new Date(2020, 0, num, 12), 'do');
-}
-
-function prettyDayName(day) {
+function prettyDayName(day, locale) {
   let days = {
-    SU: 'Sunday',
-    MO: 'Monday',
-    TU: 'Tuesday',
-    WE: 'Wednesday',
-    TH: 'Thursday',
-    FR: 'Friday',
-    SA: 'Saturday'
+    SU: new Date('2020-01-06'),
+    MO: new Date('2020-01-07'),
+    TU: new Date('2020-01-08'),
+    WE: new Date('2020-01-09'),
+    TH: new Date('2020-01-10'),
+    FR: new Date('2020-01-11'),
+    SA: new Date('2020-01-12')
   };
-  return days[day];
+  return Intl.DateTimeFormat(locale, { weekday: 'long' }).format(days[day]);
+}
+
+function formatMonthAndDay(month, i18n) {
+  let parts = Intl.DateTimeFormat(i18n.resolvedLanguage, {
+    month: 'short',
+    day: 'numeric'
+  }).formatToParts(monthUtils.parseDate(month));
+  let dayPart = parts.find(p => p.type === 'day');
+  dayPart.value = i18n.t('general.ordinal', {
+    count: monthUtils.parseDate(month).getDate(),
+    ordinal: true
+  });
+  return parts.map(part => part.value).join('');
 }
 
-export function getRecurringDescription(config) {
+export function getRecurringDescription(config, i18n) {
```

Comment:
This living in `loot-core` but relying on locale files in `desktop-client` is a bit icky and results in `i18n-parser` not being able to find this, breaking CI.

We're going to have to handle this more intelligently in future (due to `loot-design`) but as this function only gets used in `desktop-client` I think we can just move this there for now.